src/js/msp/MSPHelper.js

```diff
@@ -785,6 +785,37 @@ MspHelper.prototype.process_data = function(dataHandler) {
                     };
                     MODE_RANGES.push(modeRange);
                 }
+
+                // MODE_RANGES_EXTRA isn't support before 1.42
+                if (semver.lt(CONFIG.apiVersion, "1.42.0")){
```

Comment:
The plan is to add this for 4.0, which will be API version 1.41.

Also, please use a space before `{`.

I probably wouldn't stick filling in the defaults into the processing of `MSP_MODE_RANGES`, this is a bit unintuitive. If the adjustment range tab needs default values for `MODE_RANGES_EXTRA`, then the tab should really supply them if `MSP_MODE_RANGES_EXTRA` isn't called.

Reply:
Ah, I get it. Should have checked that the current release was 1.40.

Reply:
And I think you meant auxiliary tab. Yes, it needs default values of `MSP_MODE_RANGES`, and after having worked in the tab code a bit now, I think I have a better understanding of what I should have done. I'll make those changes.

Reply:
If I'm reading the code right, it looks like `MODE_RANGES_EXTRA` support needs to be added to backup_restore.js, correct? I can see a call there that would also require default values if the api doesn't support it. I'll test this.

Reply:
Yes please, add it to `backup_restore.js`.

Reply:
@mikeller: I added default code to `backup_restore.js`. However, I did not make changes to `migrate()` in `backup_restore.js`. It looked like there were some issues with it and didn't look like it had been updated in a while.